src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooService.java

```diff
@@ -339,6 +346,44 @@ public List<BuildLogEntry> getLatestBuildLogs(ProgrammingSubmission programmingS
         return buildLogEntries;
     }
 
+    @Override
+    public void extractAndPersistBuildLogStatistics(ProgrammingSubmission programmingSubmission, ProgrammingLanguage programmingLanguage, ProjectType projectType,
+            List<BuildLogEntry> buildLogEntries) {
+        if (buildLogEntries.isEmpty()) {
+            // No logs received -> Do nothing
+            return;
+        }
+
+        if (programmingLanguage != ProgrammingLanguage.JAVA) {
+            // Not supported -> Do nothing
+            return;
+        }
+
+        ZonedDateTime jobStarted = getTimestampForLogEntry(buildLogEntries, "started building on agent");
+        ZonedDateTime agentSetupCompleted = getTimestampForLogEntry(buildLogEntries, "Executing build");
+        ZonedDateTime testsStarted = getTimestampForLogEntry(buildLogEntries, "Starting task 'Tests'");
+        ZonedDateTime testsFinished = getTimestampForLogEntry(buildLogEntries, "Finished task 'Tests' with result");
+        ZonedDateTime scaStarted = getTimestampForLogEntry(buildLogEntries, "Starting task 'Static Code Analysis'");
+        ZonedDateTime scaFinished = getTimestampForLogEntry(buildLogEntries, "Finished task 'Static Code Analysis'");
+        ZonedDateTime jobFinished = getTimestampForLogEntry(buildLogEntries, "Finished building");
+
+        Integer dependenciesDownloadedCount = null;
+
+        // If the projectType is null, this is an old (Maven-only) exercise
+        if (projectType == null || projectType.isMaven()) {
+            // Not supported for GRADLE projects
+            dependenciesDownloadedCount = countMatchingLogs(buildLogEntries, "Downloaded from");
+        }
+
+        var agentSetupDuration = new BuildLogStatisticsEntry.BuildJobPartDuration(jobStarted, agentSetupCompleted);
```

Comment:
If you want to, you can add a static import for the Record, so you don't have to type "BuildLogStatisticsEntry." infant of it

Reply:
Thanks, I changed it.